src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1548,25 +1547,15 @@ protected virtual void DoAdditionalInstantiatedTypeProcessing (TypeDefinition ty
 
 		TypeDefinition GetDebuggerAttributeTargetType (CustomAttribute ca, AssemblyDefinition asm)
 		{
-			TypeReference targetTypeReference = null;
 			foreach (var property in ca.Properties) {
-				if (property.Name == "Target") {
-					targetTypeReference = (TypeReference) property.Argument.Value;
-					break;
-				}
+				if (property.Name == "Target")
+					return ((TypeReference) property.Argument.Value)?.Resolve ();
 
-				if (property.Name == "TargetTypeName") {
-					if (TypeNameParser.TryParseTypeAssemblyQualifiedName ((string) property.Argument.Value, out string typeName, out string assemblyName)) {
-						if (string.IsNullOrEmpty (assemblyName))
-							targetTypeReference = asm.MainModule.GetType (typeName);
```

Comment:
If the type name is not assembly qualified, this will resolve it against the current assembly.
But the new code will search all assemblies and return the first hit. For things like Nullable attributes which are added into each assembly - the behavior will suddenly be very different.

Reply:
@vitek-karas I made changes to this so that we honor the old (current) behavior.

@MichalStrehovsky I can change this back if it's too risky. I felt comfortable changing this since it seems that any type name that returns true when passed to `TryParseTypeAssemblyQualifiedName` should be correctly parsed by the new type name parser.